src/main/scala/com/campudus/tableaux/cache/CacheVerticle.scala

```diff
@@ -0,0 +1,202 @@
+    def createCache() = CacheBuilder
+      .newBuilder()
+      .expireAfterAccess(config().getLong("expireAfterAccess", 3600l).toLong, TimeUnit.SECONDS)
+      .maximumSize(config().getLong("maximumSize", 10000l).toLong)
+      .recordStats()
+      .build[String, Object]
+
+    caches.get((tableId, columnId)) match {
+      case Some(cache) => cache
+      case None =>
+        val cache = ScalaCache(GuavaCache(createCache()))
+        caches.put((tableId, columnId), cache)
+        cache
+    }
+  }
+
+  private def removeCache(tableId: TableId, columnId: ColumnId): Unit = {
+    caches.remove((tableId, columnId))
+  }
+
+  private def messageHandlerSet(message: Message[JsonObject]): Unit = {
+    val obj = message.body()
+
+    val value = obj.getValue("value")
+
+    (for {
+      tableId <- Option(obj.getLong("tableId")).map(_.toLong)
+      columnId <- Option(obj.getLong("columnId")).map(_.toLong)
+      rowId <- Option(obj.getLong("rowId")).map(_.toLong)
+    } yield (tableId, columnId, rowId)) match {
+      case Some((tableId, columnId, rowId)) =>
+
+        implicit val scalaCache = getCache(tableId, columnId)
+        put(rowId)(value)
+          .map({
+            _ =>
+              val reply = Json.obj(
+                "tableId" -> tableId,
+                "columnId" -> columnId,
+                "rowId" -> rowId
+              )
+
+              message.reply(reply)
+          })
+
+      case None =>
+        logger.error("Message invalid: Fields (tableId, columnId, rowId) should be a Long")
+        message.fail(INVALID_MESSAGE, "Message invalid: Fields (tableId, columnId, rowId) should be a Long")
+    }
+  }
+
+  private def messageHandlerRetrieve(message: Message[JsonObject]): Unit = {
+    val obj = message.body()
+
+    (for {
+      tableId <- Option(obj.getLong("tableId")).map(_.toLong)
+      columnId <- Option(obj.getLong("columnId")).map(_.toLong)
+      rowId <- Option(obj.getLong("rowId")).map(_.toLong)
+    } yield (tableId, columnId, rowId)) match {
+      case Some((tableId, columnId, rowId)) =>
+
+        implicit val scalaCache = getCache(tableId, columnId)
+
+        get[AnyRef, NoSerialization](rowId)
+          .map({
+            case Some(value) =>
+              val reply = Json.obj(
+                "tableId" -> tableId,
+                "columnId" -> columnId,
+                "rowId" -> rowId,
+                "value" -> value
+              )
+
+              message.reply(reply)
+            case None =>
+              logger.warn(s"messageHandlerRetrieve $tableId, $columnId, $rowId not found")
+              message.fail(NOT_FOUND_FAILURE, "Not found")
+          })
+
+      case None =>
+        logger.error("Message invalid: Fields (tableId, columnId, rowId) should be a Long")
+        message.fail(INVALID_MESSAGE, "Message invalid: Fields (tableId, columnId, rowId) should be a Long")
+    }
+  }
+
+  private def messageHandlerInvalidate(message: Message[JsonObject]): Unit = {
```

Comment:
This method looks like it does two things at once - invalidating a complete column and invalidating a single cell?